src/Parser/Warlock/Affliction/Modules/Talents/MaleficGrasp.js

```diff
@@ -42,46 +47,81 @@ class MaleficGrasp extends Analyzer {
     if (!AFFECTED_ABILITIES.includes(spellId)) {
       return;
     }
+
     const target = this.enemies.getEntity(event);
     if (!target) {
       return;
     }
-    const buffedByDrain = target.hasBuff(SPELLS.DRAIN_SOUL.id, event.timestamp);
-    if (!buffedByDrain) {
-      return;
+
+    const drained = target.hasBuff(SPELLS.DRAIN_SOUL.id, event.timestamp);
+    if (AFFECTED_ABILITIES.includes(spellId)) {
```

Comment:
I assume this is supposed to be `UNSTABLE_AFFLICTION_DEBUFF_IDS` like the other modules?

Reply:
No, this is correct, there's an array of affected abilities (apart from UA it also contains Agony, Corruption and Siphon Life).

Reply:
Isn't it only UA ticks you want to count though? The statbox label says "UA ticks buffed by Malefic Grasp" so I'd assume one of them needs to be changed.

Reply:
That's a good point. The stat box is oriented around buffed UA ticks, which matters the most, but the tooltip also shows total damage contributed by the talent. Might be more sensible to split that into two modules but I can't help but think that's a bit redundant.

Reply:
I think it's fine to just display either percent of UA ticks buffed or DPS contributed, which ever you think is more important, and then show the other in the tooltip. If you show the DPS contributed you could also add the percent of buffed dot ticks for each of the ticks individually in the tooltip. 

Alternatively you could also make a module for Drain Soul that shows time spent channeling as that should also give a pretty good indication of how many Agony/Corruption ticks were buffed by MG.

Reply:
I would prefer the first approach which is what's currently impletented if I understood you correctly - percentage of buffed UA ticks (which is more important when running the talent), and in the tooltip damage done (+ % of total damage done), and then how much damage it contributed for each of the affected dots.

![image](https://user-images.githubusercontent.com/5068584/35442006-eb8099ae-02a5-11e8-8b50-6836b38dab8e.png)

Reply:
Just remember to change it so it's only the UA ticks being counted! :D

Reply:
Woops, I misunderstood you in the beginning.. Aggressive refactoring 😅